simpleBuildAgainstTrilinos: install testing issues with Cuda builds

## Bug Report
After merge of #11863 I am encountering problems when testing a Trilinos install with the `simpleBuildAgainstTrilinos` demo with Cuda builds.

I'm adding the @trilinos/tribits label for now, can re-label as needed, @bartlettroscoe 

Here are some notes, reproducer details will follow below

**`simpleBuildAgainstTrilinos` configuration output:** (leading to failed compilation)
```
20:19:51 -- The C compiler identification is GNU 8.5.0
20:19:51 -- The CXX compiler identification is GNU 8.5.0
20:19:51 -- Detecting C compiler ABI info
20:19:51 -- Detecting C compiler ABI info - done
20:19:51 -- Check for working C compiler: /usr/bin/cc - skipped
20:19:51 -- Detecting C compile features
20:19:51 -- Detecting C compile features - done
20:19:51 -- Detecting CXX compiler ABI info
20:19:51 -- Detecting CXX compiler ABI info - done
20:19:51 -- Check for working CXX compiler: /usr/bin/c++ - skipped
20:19:51 -- Detecting CXX compile features
20:19:51 -- Detecting CXX compile features - done
20:19:51 -- Found CUDAToolkit: /home/projects/ppc64le-pwr9-nvidia/spack-installs/cuda/11.2.2/gcc/8.3.1/base/uflzqtv/include (found version "11.2.152") 
20:19:52 -- Looking for pthread.h
20:19:52 -- Looking for pthread.h - found
20:19:52 -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
20:19:52 -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
20:19:52 -- Looking for pthread_create in pthreads
20:19:52 -- Looking for pthread_create in pthreads - not found
20:19:52 -- Looking for pthread_create in pthread
20:19:52 -- Looking for pthread_create in pthread - found
20:19:52 -- Found Threads: TRUE  
20:19:53 -- Enabled Kokkos devices: CUDA;SERIAL
20:19:53 
20:19:53 Found Trilinos!  Here are the details: 
20:19:53    Trilinos_DIR = /home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/Build/install/lib/cmake/Trilinos
20:19:53    Trilinos_VERSION = 14.1
20:19:53    Trilinos_PACKAGE_LIST = Adelus;Phalanx;Tpetra;TpetraCore;TpetraTSQR;Epetra;Sacado;KokkosKernels;Teuchos;TeuchosKokkosComm;TeuchosKokkosCompat;TeuchosRemainder;TeuchosNumerics;TeuchosComm;TeuchosParameterList;TeuchosParser;TeuchosCore;Kokkos
20:19:53    Trilinos_LIBRARIES = Teuchos::all_libs;Tpetra::all_libs
20:19:53    Trilinos_INCLUDE_DIRS = /home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/Build/install/include
20:19:53    Trilinos_TPL_LIST = 
20:19:53    Trilinos_TPL_LIBRARIES = 
20:19:53    Trilinos_BUILD_SHARED_LIBS = ON
20:19:53 End of Trilinos details
20:19:53 
20:19:53 -- Configuring done
20:19:53 -- Generating done
20:19:53 -- Build files have been written to: /home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/Build/app-test
```

The configuration above leads to compilation errors, as g++ is being picked up as the compiler rather than nvcc_wrapper through the OMPI_CXX environment variable

**Compilation error:**
```
20:19:53 [ 50%] Building CXX object CMakeFiles/MyApp.dir/app.cpp.o
20:19:53 c++: error: unrecognized command line option ‘-Wext-lambda-captures-this’
20:19:53 c++: error: unrecognized command line option ‘-arch=sm_70’
20:19:53 make[2]: *** [CMakeFiles/MyApp.dir/build.make:76: CMakeFiles/MyApp.dir/app.cpp.o] Error 1
20:19:53 make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/MyApp.dir/all] Error 2
20:19:53 make: *** [Makefile:101: all] Error 2
```


My cmake line for configuring `simpleBuildAgainstTrilinos` is as follows, which worked prior to the #11863 updates:
```
cmake  -DCMAKE_PREFIX_PATH="$TRILINOS_INSTALL" $TRILINOS_DIR/demos/simpleBuildAgainstTrilinos
```


From comparison, this is the configuration output from the previous round of passing tests:
```
20:17:24 -- Found CUDAToolkit: /home/projects/ppc64le-pwr9-nvidia/spack-installs/cuda/11.2.2/gcc/8.3.1/base/uflzqtv/include (found version "11.2.152") 
20:17:24 CMake Warning at /projects/ppc64le/spack-installs/cmake/3.23.1/gcc/7.2.0/base/knlpf2j/share/cmake-3.23/Modules/FindCUDAToolkit.cmake:919 (message):
20:17:24   Could not find librt library, needed by CUDA::cudart_static
20:17:24 Call Stack (most recent call first):
20:17:24   /projects/ppc64le/spack-installs/cmake/3.23.1/gcc/7.2.0/base/knlpf2j/share/cmake-3.23/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
20:17:24   /home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/Build/install/lib/external_packages/CUDA/CUDAConfig.cmake:14 (find_dependency)
20:17:24   /home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/Build/install/lib/cmake/Kokkos/KokkosConfig.cmake:193 (include)
20:17:24   /home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/Build/install/lib/cmake/TeuchosCore/TeuchosCoreConfig.cmake:175 (include)
20:17:24   /home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/Build/install/lib/cmake/Teuchos/TeuchosConfig.cmake:172 (include)
20:17:24   /home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/Build/install/lib/cmake/Trilinos/TrilinosConfig.cmake:123 (include)
20:17:24   CMakeLists.txt:16 (find_package)
20:17:24 
20:17:24 
20:17:25 -- Enabled Kokkos devices: CUDA;SERIAL
20:17:25 
20:17:25 Found Trilinos!  Here are the details: 
20:17:25    Trilinos_DIR = /home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/Build/install/lib/cmake/Trilinos
20:17:25    Trilinos_VERSION = 14.1
20:17:25    Trilinos_PACKAGE_LIST = Adelus;Phalanx;Tpetra;TpetraCore;TpetraTSQR;Epetra;Sacado;KokkosKernels;Teuchos;TeuchosKokkosComm;TeuchosKokkosCompat;TeuchosRemainder;TeuchosNumerics;TeuchosComm;TeuchosParameterList;TeuchosParser;TeuchosCore;Kokkos
20:17:25    Trilinos_LIBRARIES = Teuchos::all_libs;Tpetra::all_libs
20:17:25    Trilinos_INCLUDE_DIRS = /home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/Build/install/include
20:17:25    Trilinos_TPL_LIST = 
20:17:25    Trilinos_TPL_LIBRARIES = 
20:17:25    Trilinos_BUILD_SHARED_LIBS = ON
20:17:25 End of Trilinos details
20:17:25 
20:17:25 -- The C compiler identification is GNU 8.5.0
20:17:25 -- Detecting C compiler ABI info
20:17:25 -- Detecting C compiler ABI info - done
20:17:25 -- Check for working C compiler: /home/projects/ppc64le-pwr9-nvidia/spack-installs/openmpi/4.1.1/gcc/8.3.1/base/elujn3i/bin/mpicc - skipped
20:17:25 -- Detecting C compile features
20:17:25 -- Detecting C compile features - done
20:17:27 -- The CXX compiler identification is GNU 8.5.0
20:17:27 -- Detecting CXX compiler ABI info
20:17:28 -- Detecting CXX compiler ABI info - done
20:17:29 -- Check for working CXX compiler: /home/projects/ppc64le-pwr9-nvidia/spack-installs/openmpi/4.1.1/gcc/8.3.1/base/elujn3i/bin/mpicxx - skipped
20:17:29 -- Detecting CXX compile features
20:17:29 -- Detecting CXX compile features - done
20:17:29 -- The Fortran compiler identification is GNU 8.5.0
20:17:29 -- Detecting Fortran compiler ABI info
20:17:29 -- Detecting Fortran compiler ABI info - done
20:17:29 -- Check for working Fortran compiler: /home/projects/ppc64le-pwr9-nvidia/spack-installs/openmpi/4.1.1/gcc/8.3.1/base/elujn3i/bin/mpifort - skipped
20:17:29 -- Configuring done
20:17:29 -- Generating done
20:17:29 -- Build files have been written to: /home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/Build/app-test
```

In case this is relevant, these failures began showing in nightly tests against Kokkos Core+Kernels develop branches where I use source override to build using the source code from their respective repos (rather than the Trilinos packages). I still need to reproduce with just Trilinos' develop branch


### Steps to Reproduce
1. SHA1:  46ded0b3a41bab08e4b5476192f21e50741a653c
2.Reproducer using sems modules:
```bash
# repos and sym-link for source override
git clone -b develop https://github.com/trilinos/Trilinos.git
git clone -b develop https://github.com/kokkos/kokkos.git
git clone -b develop https://github.com/kokkos/kokkos-kernels.git

# Symbolic link to kokkos and kokkos-kernels repos in Trilinos source directory for source override
cd Trilinos
ln -s <path-to-your-repo>/kokkos kokkos
ln -s <path-to-your-repo>/kokkos-kernels kokkos-kernels

cd $HOME
mkdir -p build
cd build

# modules and env
sh /projects/sems/modulefiles/utils/sems-v2-modules-init.sh
module load sems-cmake/3.24.3 sems-gcc/8.3.0 sems-cuda/11.4.2 sems-openmpi/4.0.5 sems-python/3.7.9 sems-ninja/1.10.1
export OMPI_CXX=$KOKKOS_PATH/bin/nvcc_wrapper

# configure
cmake  \
  -GNinja  \
  -DCMAKE_CXX_STANDARD=17 \
  -DCMAKE_INSTALL_PREFIX="${TRILINOS_INSTALL}" \
  -DTPL_ENABLE_MPI=ON \
  -DTrilinos_ENABLE_TESTS=OFF \
  -DTrilinos_ENABLE_Kokkos=ON \
   -DKokkos_ENABLE_CUDA=ON \
   -DKokkos_ENABLE_CUDA_UVM=ON \
   -DKokkos_ENABLE_CUDA_LAMBDA=ON \
  -DTrilinos_ENABLE_KokkosKernels=ON \
   -DKokkosKernels_INST_MEMSPACE_CUDAUVMSPACE=ON \
  -DTrilinos_ENABLE_Tpetra=ON \
  -DKokkos_SOURCE_DIR_OVERRIDE:STRING=kokkos \
  -DKokkosKernels_SOURCE_DIR_OVERRIDE:STRING=kokkos-kernels \
$TRILINOS_DIR

# install Trilinos
make install -j8

# test app build against trilinos install
mkdir -p app-test
cd app-test
cmake -DCMAKE_PREFIX_PATH=${TRILINOS_INSTALL} ${TRILINOS_DIR}/demos/simpleBuildAgainstTrilinos
make
ctest
```

> @bartlettroscoe here is my LastConfigure_*.log file (I was distracted and looked in the wrong build dir before)
> 
> [LastConfigure_20230608-1858.log](https://github.com/trilinos/Trilinos/files/11692557/LastConfigure_20230608-1858.log)

@ndellingwood, if you look at that file, you will see:

```
Processing enabled external package/TPL: Netcdf (enabled by SEACASNemesis, disable with -DTPL_ENABLE_Netcdf=OFF)
-- Using find_package(Netcdf ...) ...
CMake Error at cmake/tribits/common_tpls/find_modules/FindNetCDF.cmake:163 (message):
  Can not locate NetCDF include directory
Call Stack (most recent call first):
  cmake/tribits/common_tpls/FindTPLNetcdf.cmake:92 (find_package)
  cmake/tribits/core/package_arch/TribitsProcessEnabledTpls.cmake:219 (include)
  cmake/tribits/core/package_arch/TribitsProcessEnabledTpls.cmake:169 (tribits_process_enabled_tribits_find_tpl_mod_file)
  cmake/tribits/core/package_arch/TribitsProcessEnabledTpls.cmake:104 (tribits_process_enabled_standard_tpl)
  cmake/tribits/core/package_arch/TribitsProjectImpl.cmake:193 (tribits_process_enabled_tpls)
  cmake/tribits/core/package_arch/TribitsProject.cmake:92 (tribits_project_impl)
  CMakeLists.txt:126 (TRIBITS_PROJECT)


CMake Error at cmake/tribits/common_tpls/find_modules/FindNetCDF.cmake:276 (message):
```

This looks to be an env problem.  How is it that `find_package(Netcdf ...)` does not find `Netcdf` when the initial configure did?

Can you please provide the STDOUT from the from-scratch configure?

```
$ rm -r CMake*
$ ./do-configure &> configure.out
```

Then, what happens when you just run:

```
$ cmake . &> reconfigure.out
```

instead of `make dashboard`?  What is inside of that STDOUT file `reconfigure.out`?

--- a/demos/simpleBuildAgainstTrilinos/README.md
+++ b/demos/simpleBuildAgainstTrilinos/README.md
@@ -17,6 +17,8 @@ To run this example, do the following steps.
     -D CMAKE_INSTALL_PREFIX=<tril-install-prefix> \
     -D TPL_ENABLE_MPI=ON \
     -D Trilinos_ENABLE_Tpetra=ON \
+    -D CMAKE_C_COMPILER=<c-compiler> \
+    -D CMAKE_CXX_COMPILER=<cxx-compiler> \    
     [other options] \
     <trilinos-src>
 ```
@@ -42,8 +44,15 @@ This will put a file called `TrilinosConfig.cmake` under a subdirectory of
   $ cd <app-build-dir>/
   $ cmake \
     -D CMAKE_PREFIX_PATH=<tril-install-prefix> \
+    -D CMAKE_C_COMPILER=<c-compiler> \
+    -D CMAKE_CXX_COMPILER=<cxx-compiler> \    
     <trilinos-src>/demos/simpleBuildAgainstTrilinos
 ```
+
+NOTE: To configure `simpleBuildAgainstTrilinos` against the installed Trilinos
+one must pass in compatible compilers used to build Trilinos. For CUDA builds,
+this means using `nvcc_wrapper` (e.g. using OpenMPI with `-D CMAKE_CXX_COMPILER=mpicxx`
+and setting environment variable `export OMPI_CXX=<base-path>/nvcc_wrapper`.)
 
 NOTE: To use the alternative `CMakeLists.txt` file that calls
 `find_package(Tpetra)` and links to `Tpetra::all_libs` instead of

